src/defold/support/ScriptMacro.hx

```diff
@@ -368,11 +370,12 @@ private class Glue {
             case PInt: '0';
             case PFloat: '0.0';
             case PQuaternion: 'vmath.quat()';
-            case PAtlasResourceReference
-                | PFontResourceReference
-                | PMaterialResourceReference
-                | PTextureResourceReference
-                | PTileSourceResourceReference: throw new Error('Property of type ${Std.string(type).substr(1)} cannot have an empty value.', Context.currentPos());
+			case PBufferResourceReference: 'resource.buffer()';
+			case PAtlasResourceReference: 'resource.atlas()';
+			case PFontResourceReference: 'resource.font()';
+			case PMaterialResourceReference: 'resource.material()';
+			case PTextureResourceReference: 'resource.texture()';
+			case PTileSourceResourceReference: 'resource.tile_source()';
```

Comment:
Indentation needs fixing, the repository is using 4 spaces.

Reply:
Also move `PBufferResourceReference` to the end so that the order is consistent with the delcaration.